client/ladder/action-creators.ts

```diff
@@ -70,7 +113,57 @@ export function searchRankings(
     )
 
     // Don't update the state if we aren't the last request outstanding
-    if (fetchTime >= (lastFetchTimeByMatchmakingType.get(matchmakingType) ?? 0)) {
+    if (fetchTime >= (lastSearchTimeByMatchmakingTypeAndSeasonId.get(cacheKey) ?? 0)) {
+      lastSearchTimeByMatchmakingTypeAndSeasonId.set(cacheKey, fetchTime)
+      lastSearchQueryByMatchmakingTypeAndSeasonId.set(cacheKey, searchQuery)
+
+      dispatch({
+        type: '@ladder/searchRankings',
+        payload: result,
+        meta: { matchmakingType, searchQuery, fetchTime: new Date() },
+      })
+
+      dispatch({
+        type: '@matchmaking/getCurrentMatchmakingSeason',
+        payload: result.season,
+      })
+    }
+  })
+}
+
+export function searchPreviousSeasonRankings(
+  seasonId: SeasonId,
+  matchmakingType: MatchmakingType,
+  searchQuery: string,
+  spec: RequestHandlingSpec<void>,
+): ThunkAction {
+  return abortableThunk(spec, async dispatch => {
+    const fetchTime = performance.now()
+    const cacheKey = makeMatchmakingTypeAndSeasonId(matchmakingType, seasonId)
+    const lastSearchTime = lastSearchTimeByMatchmakingTypeAndSeasonId.get(cacheKey)
+    const lastSearchQuery = lastSearchQueryByMatchmakingTypeAndSeasonId.get(cacheKey)
```

Comment:
If search time and query are always set together, they should probably be stored in the same map instead of separately